core/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TasksIT.java

```diff
@@ -406,6 +408,17 @@ public void testTasksListWaitForNoTask() throws Exception {
         assertThat(waitResponseFuture.get().getTasks(), emptyCollectionOf(TaskInfo.class));
     }
 
+    public void testTasksWaitForAllTask() throws Exception {
+        // Spin up a request to wait for all tasks in the cluster
+        ListTasksResponse response = client().admin().cluster().prepareListTasks().setWaitForCompletion(true)
```

Comment:
I'd probably use the words "infinite loop" somewhere in a comment explaining this.